server/aws/client_cache.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package aws
+
+import (
+	sdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+
+	"github.com/pkg/errors"
+)
+
+// DefaultRegion describes default region in aws
+const DefaultRegion = "us-east-2"
+
+type clientCache struct {
+	store CredentialStoreService
+	cache map[string]*Client // appID - Client
+}
+
+func newClientCache(store CredentialStoreService) (*clientCache, error) {
+	cache := make(map[string]*Client)
+	list, err := store.GetAppList()
+	if err != nil {
+		return nil, errors.Wrap(err, "can't get credentials")
+	}
+	cc := &clientCache{
+		store: store,
+		cache: cache,
+	}
+	for _, appID := range list {
+		if _, err := cc.create(appID); err != nil {
```

Comment:
Is it better to create all the clients on init, or it is better to create the clients on demand? (i.e. in the get function if the client does not exist)